airflow_dag_samples/graph_data_dag/sdk_calls.py

```diff
@@ -0,0 +1,230 @@
+                            re_relationships = relation_results.data("relationship")
+                            extracted_re_relationships = self.extract_node_properties(re_relationships)
+                            for relationship in extracted_re_relationships:
+                                relations.append({"obj": {"entity": m2.get("value"), "type": self.convert_labels(m2.get("ne")[0])}, "sub": {"entity": m1.get("value"), "type": self.convert_labels(m1.get("ne")[0])}, "relation": relationship.get("type")})
+                    sentence_text = sentence.get("sentence").get("sentence")
+                    sentence_text = self.add_whitespace_to_punctuation(sentence_text)
+                    mentions = [mention.get("mention") for mention in extracted_mention_nodes]
+                    sentence_json["sentence"] = sentence_text
+                    if translated_text:
+                        sentence_json["translations"] = {}
+                        sentence_json["translations"][orig_language] = extracted_translation_nodes[0].get("origSentence").get("sentence")
+                    sentence_json["keyphrases"] = [keyphrase.get("keyphrase").get("keyphraseId") for keyphrase in extracted_keyphrase_nodes]
+                    sentence_json["named_entity"] = self.get_ner_list(sentence_text, mentions)
+                    sentence_json["relations"] = relations
+
+                    with open(os.path.join('/volumes', filename), 'a+', encoding='utf-8') as f:
+                        f.write(json.dumps(sentence_json))
+                        f.write('\n')
+
+        return filename
+
+    def save_dataset(self, filename: str):
+        sdk = GraphGridSdk()
+
+        def read_by_line(dataset_filepath):
+            infile = open(
+                dataset_filepath,
+                'r', encoding='utf8')
+            for line in infile:
+                yield line.encode()
+
+        filepath = os.path.join('/volumes', filename)
+        yield_function = read_by_line(filepath)
+        dataset_response: SaveDatasetResponse = sdk.save_dataset(yield_function,
+                                                                 filename)
+        if dataset_response.status_code != 200:
+            raise Exception("Failed to save dataset: ",
+                            dataset_response.exception)
+
+        return dataset_response.datasetId
+
+    def train_and_promote(self, models_to_train: list,
+                          datasetId: str,
+                          no_cache: typing.Optional[bool],
+                          gpu: typing.Optional[bool], autopromote: bool):
+        sdk = GraphGridSdk()
+
+        def success_handler(args):
+            return
+
+        def failed_handler(args):
+            return
+
+        sdk.nmt_train_pipeline(models_to_train, datasetId, no_cache, gpu,
+                               autopromote, success_handler, failed_handler)
+
+
+def main():
+    fire.Fire(Pipeline)
+
+
+if __name__ == '__main__':
+    main()
+
```

Comment:
Noting that we'll probably want to clean this up and remove all these extraneous newlines @alexiudice